code/frameworks/angular/src/client/angular-beta/utils/PropertyExtractor.ts

```diff
@@ -0,0 +1,123 @@
+import { CommonModule } from '@angular/common';
+import { NgModuleMeta } from '@angular/compiler-cli/src/ngtsc/metadata';
+import { NgModule, Provider } from '@angular/core';
+import { NgModuleMetadata } from '../../types';
+import { isDeclarable, isStandaloneComponent } from './NgComponentAnalyzer';
+import { isComponentAlreadyDeclared } from './NgModulesAnalyzer';
+
+/**
+ * Flattens Array Deeply
+ *
+ * - Executes recursively
+ *
+ * @date `1/24/2023 - 12:58:03 PM`
+ * @author `sheriffMoose`
```

Comment:
I'm not sure if this was added by your IDE or manually, but usually, for readability reasons, we don't add author information to each function or method. Second, this information has to be updated every time another developer touches the code. Let's remove this.